keep-ui/utils/hooks/useSearchAlerts.ts

```diff
@@ -0,0 +1,36 @@
+import useSWR, { SWRConfiguration } from "swr";
+import { AlertDto } from "app/alerts/models";
+import { useSession } from "next-auth/react";
+import { getApiURL } from "utils/apiUrl";
+import { fetcher } from "utils/fetcher";
+import { useDebouncedValue } from "./useDebouncedValue";
+import { RuleGroupType, formatQuery } from "react-querybuilder";
+
+export const useSearchAlerts = (
+  args: { query: RuleGroupType; timeframe: number },
+  options?: SWRConfiguration
+) => {
+  const apiUrl = getApiURL();
+  const { data: session } = useSession();
+
+  const [debouncedArgs] = useDebouncedValue(args, 2000);
```

Comment:
Is a 2 seconds delay enough?

Reply:
think so, let's adjust when needed if not